karabo_data/geometry2.py

```diff
@@ -0,0 +1,483 @@
+
+        pc = PatchCollection(rects, facecolor=(0.75, 1., 0.75), edgecolor=None)
+        ax.add_collection(pc)
+
+        ax.hlines(0, -100, +100, colors='0.75', linewidths=2)
+        ax.vlines(0, -100, +100, colors='0.75', linewidths=2)
+
+        ax.set_title('AGIPD-1M detector geometry')
+        return fig
+
+    def position_all_modules(self, data):
+        """Assemble data from this detector according to where the pixels are.
+
+        This performs interpolation, which is very slow.
+        To efficiently copy the data into a 2D array, first use the snap()
+        method to get a pixel-aligned approximation of the geometry.
+
+        Parameters
+        ----------
+
+        data : ndarray
+          The three dimensions should be channelno, pixel_ss, pixel_fs
+          (lengths 16, 512, 128). ss/fs are slow-scan and fast-scan.
+
+        Returns
+        -------
+        out : ndarray
+          Array with the one dimension fewer than the input.
+          The last two dimensions represent pixel y and x in the detector space.
+        centre : ndarray
+          (x, y) pixel location of the detector centre in this geometry.
+        """
+        assert data.shape == (16, 512, 128)
+        size_xy, centre = self._plotting_dimensions()
+        size_yx = size_xy[::-1]
+        tmp = np.empty((16 * 8,) + size_yx, dtype=data.dtype)
+
+        for i, (module, mod_data) in enumerate(zip(self.modules, data)):
+            tiles_data = np.split(mod_data, 8)
+            for j, (tile, tile_data) in enumerate(zip(module, tiles_data)):
+                # We store (x, y, z), but numpy indexing, and hence affine_transform,
+                # work like [y, x]. Rearrange the numbers:
+                fs_vec_yx = tile.fs_vec[:2][::-1]
+                ss_vec_yx = tile.ss_vec[:2][::-1]
+
+                # Offset by centre to make all coordinates positive
+                corner_pos = (tile.corner_pos[:2] + centre)
+                corner_pos_yx = corner_pos[::-1]
+
+                # Make the rotation matrix
+                rotn = np.stack((ss_vec_yx, fs_vec_yx), axis=-1)
+
+                # affine_transform takes a mapping from *output* to *input*.
+                # So we reverse the forward transformation.
+                transform = np.linalg.inv(rotn)
+                offset = np.dot(rotn, corner_pos_yx)  # this seems to work, but is it right?
+
+                affine_transform(tile_data, transform, offset=offset, cval=np.nan,
+                                 output_shape=size_yx, output=tmp[i * 8 + j])
+
+        # Silence warnings about nans - we expect gaps in the result
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", category=RuntimeWarning)
+            out = np.nanmax(tmp, axis=0)
+
+        return out, centre
+
+    def _plotting_dimensions(self):
+        """Calculate appropriate dimensions for plotting assembled data
+
+        Returns (size_x, size_y), (centre_x, centre_y)
```

Comment:
I think this would be better if it would be (size_y, size_x), (centre_y, centre_x) to be consistent with the _plotting_dimension method in the snapped class. I've already had some confusion/issues with it

Reply:
Yeah, I'll see if I can switch everything to `y, x`. It's an unfortunate discrepancy between accessing the array by `row, column` and referring to plotting coordinates as `x, y`.